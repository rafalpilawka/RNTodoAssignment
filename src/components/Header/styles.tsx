import {StyleSheet} from 'react-native';
import normalize from 'react-native-normalize';
import {colors} from 'src/constants/platformSpecificColors';

export const styles = StyleSheet.create({
  headerContainer: {
    height: normalize(50),
    width: '100%',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomWidth: 1,
    backgroundColor: colors!.color,
    marginBottom: normalize(15),
    paddingHorizontal: normalize(20)
  },
  row: {
    flexDirection: 'row',
  },
  titleText: {
    fontSize: normalize(20),
    fontWeight: 'bold',
    fontFamily: 'Damascus',
    color: 'white',
  },
  lifter:{paddingBottom: normalize(6),}
});
