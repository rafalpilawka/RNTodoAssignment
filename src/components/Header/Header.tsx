import React, {ReactElement} from 'react';
import {View, Text, Pressable} from 'react-native';
import {styles} from './styles';

interface Props {
  back?(): void;
  listId?: string;
}
export const Header = ({back, listId}: Props): ReactElement => {
  return (
    <View style={[styles.row, styles.headerContainer]}>
      {back ? (
        <Pressable onPress={back}>
          <Text style={[styles.titleText, styles.lifter]}>{'<'}</Text>
        </Pressable>
      ) : (
        <View />
      )}
      <Text style={styles.titleText}>
        {listId ? listId.toUpperCase() : 'INDEX OF TODOS'}
      </Text>
      <View />
    </View>
  );
};
