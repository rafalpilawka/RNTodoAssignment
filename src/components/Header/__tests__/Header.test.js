import React from 'react';
import 'react-native';
import {Pressable} from 'react-native';
import {shallow} from 'enzyme';
import {Header} from 'src/components/Header/Header';
import renderer from 'react-test-renderer';

const props = {
  back: jest.fn(),
  listId: 'Testing list id',
};

describe('Header', () => {
  let component;
  beforeEach(() => {
    component = shallow(<Header {...props} />);
    jest.clearAllMocks();
  });
  it('should match to snapshot', () => {
    const component = renderer.create(<Header {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should receive props', () => {
    const nameLabel = component.find({children: props.listId.toUpperCase()});
    expect(nameLabel).toHaveLength(1);
  });

  it(`shouldn't render back button`, () => {
    component = shallow(<Header back={false} />);
    const button = component.find(Pressable);
    expect(button).toHaveLength(0);
  });

  it('should call back button', () => {
    const button = component.find(Pressable);
    button.simulate('press');
    expect(props.back).toHaveBeenCalledTimes(1);
  });
});
