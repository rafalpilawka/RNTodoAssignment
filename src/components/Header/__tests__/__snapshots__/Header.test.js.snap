// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match to snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "alignItems": "center",
        "backgroundColor": Object {
          "semantic": Array [
            "systemBlue",
          ],
        },
        "borderBottomWidth": 1,
        "height": 100,
        "justifyContent": "space-between",
        "marginBottom": 30,
        "paddingHorizontal": 40,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Damascus",
            "fontSize": 40,
            "fontWeight": "bold",
          },
          Object {
            "paddingBottom": 12,
          },
        ]
      }
    >
      &lt;
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Damascus",
        "fontSize": 40,
        "fontWeight": "bold",
      }
    }
  >
    TESTING LIST ID
  </Text>
  <View />
</View>
`;
